File: javascript/10.js
const isPrime = (num, sieve) => {
	let idx = 0;

	for (let prime of sieve) {
		if (num % prime === 0) { return false; }
		if (Math.pow(prime, 2) > num) { return true; }
	}
};

const primeSieve = (lim) => {
	let sieve = [2];
	let num = 3;

	while (num < lim) {
		if (isPrime(num, sieve)) { sieve.push(num); }
		num += 2;
	}

	return sieve.reduce((x, y) => { return x + y; });
};

// Sum of all the primes below two million
console.log(primeSieve(2*Math.pow(10, 6)));

File: python/10.py
def is_prime(num, sieve):
	idx = 0

	for i in sieve:
		if not num % i:
			return False
		if i**2 > num:
			return True

def prime_sieve(lim):
	sieve = [2]
	num = 3

	while num < lim:
		if is_prime(num, sieve):
			sieve.append(num)
		num += 2

	return sum(sieve);

# Sum of all the primes below two million
print(prime_sieve(2*10**6))